pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -333,6 +333,12 @@ func (ks *scaler) scale(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscal
 
 	min, max := pa.ScaleBounds(asConfig)
 	initialScale := kparesources.GetInitialScale(asConfig, pa)
+	// if minNonZeroReplicas is set, use that value for initial scale
+	// note that minNonZeroReplicas will take precedence over initialScale if both are set
+	if mnzr, ok := pa.MinNonZeroReplicas(); ok && initialScale < mnzr {
+		initialScale = mnzr
```

Comment:
What's the motivation for overriding initial scale here?

ie. I'm thinking if it's explicitly set `initial-scale: 0` this breaks that

Reply:
Fair point, I wasn't thinking of initial scale=zero... 

For non-zero cases, I still slightly leans towards overriding initial scale if its smaller (I'm not sure if it makes much sense to distinguish between the first scale up and all subsequent scales), but I could easily be nudged the other direction.

Reply:
Having had another think, let's not have this mess with initial scale at all

Reply:
Having gotten around to make the changes, initial-scale=1 is a bit of a corner case (is it the default, or did someone explicitly set it to 1?). I wonder if it might be better to have min-activate-scale only apply on the 2nd, 3rd, etc. scale ups and require initial-scale to be set for the first time if you want a non-one value?